integration-tests/src/test/java/io/quarkus/fury/it/FuryTest.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.fury.it;
+
+import org.apache.fury.Fury;
+import org.apache.fury.ThreadSafeFury;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+import java.util.Map;
+
+@QuarkusTest
+public class FuryTest {
+    private static final ThreadSafeFury fury = Fury.builder()
+      .requireClassRegistration(false).buildThreadSafeFury();
```

Comment:
It need to get `Fury` from the Container by CDI like:

```java
@Inject
Fury fury;
```

Reply:
If so, who is creating Fury? Is the quarkus-fury/ or the users?

Reply:
But how do I know which options users  will set for Fury? Is is feasible to let users pass an Fury instance on build time. Another solution is we define a Factory and let users specify that factory:

```
interface FuryFactory {
  Fury build(ClassLoader classloader);
}
```

Then we use this factory to create Fury, and pass this instance to users like:
```java
@Inject
ThreadSafeFury fury;
```

Reply:
First create a FuryBuildItem on deployment module to hold a `Fury` runtime instance like:
```java
public final class FuryBuildItem extends SimpleBuildItem {
     private final RuntimeValue<Fury> value;

    public FuryBuildItem(RuntimeValue<Fury> value) {
        this.value = value;
    }

    public RuntimeValue<Fury> getFury() {
        return value;
    }
}
```

Add `FuryBuildTimeConfig.java` on runtime module for all of options we want to use like:
```java
@ConfigRoot(phase = ConfigPhase.BUILD_TIME_AND_RUN_TIME_FIXED)
@ConfigMapping(prefix = "quarkus.fury")
interface FuryBuildTimeConfig {
    /**
     * whether to use localTread;
     */
    public boolean localThread();
}
```

In `FuryProcessor` create a `BuildStep` like
```java
@BuildStep
@Record(ExecutionTime.STATIC_INIT)
public FuryBuildItem setup(FuryBuildTime config, BeanContainerBuildItem beanContainer, FuryRecorder recorder) {
    return new FruyBuildItem(recorder->createFury(config, beanContainer));
}
```

Add `FuryProducer.java` on runtime module which produces `Fury` like:
```java
@Singleton
public FuryProducer {
    private volatile Fury fury;

    public void setFury(Fury fury) {
        this.fury = fury;
    }

    @Singleton
    @Produces
    public Fury getFury() {
        return this.fury;
    }
}
```

Add `FuryRecorder.java` on runtime module:
```java
@Recorder
public class Recorder {
    public RuntimeValue<Fury> createFury(FuryBuildTimeConfig config, BeanContainer beanContainer) {
        // create the Fury instance from the config here;
        Fury fury = Fury.builder()...;

        // register to the container
       beanContainer.beanInstance(Fury.class).setFury(fury);

       return new RuntimeValue<>(fury);
    }
}
```

@chaokunyang I just write down the idea in my mind.

Reply:
How users will configure `FuryBuildTimeConfig`?

Reply:
By using `application.properties` like
```
quarkus.fury.localthread=true
quarkus.fruy.classregistration=true
...
```